troubadix/plugin.py

```diff
@@ -46,6 +45,19 @@ class Plugin(ABC):
     description: str = None
 
 
+class PreRunPlugin(Plugin):
+    """A plugin that only runs PreRun collectors"""
+
+    @staticmethod
+    @abstractmethod
+    def run(
+        pre_run_data: dict,
+        nasl_files: List[Path],
```

Comment:
```suggestion
        nasl_files: Iterable[Path],
```

Reply:
Does this match `List`? If not, it don't work here.

Reply:
@y0urself

Reply:
List are Iterables but not Iterators, thus it is okay here.